charts/jobs/templates/job.yaml

```diff
@@ -39,35 +54,39 @@ spec:
           - |
             {{- toString $spec.script | nindent 12 }}
         resources: {{- if $spec.resources}}{{ toYaml $spec.resources | nindent 10 }}{{ else }}{{ toYaml $v.resources | nindent 10 }}{{ end }}
+        {{- if $spec.secretKeyRefs }}
+        env:
+          {{- range $ref := $spec.secretKeyRefs }}
+          - name: {{ $ref.name }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ $ref.secretKeyRefName }}
+                key: {{ $ref.key }}
+          {{- end }}
+        {{- end }}
```

Comment:
I have considered this while I was going through the documentation (https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables). 

It would seem to me that in the case the user wants to reference multiple keys from the same secret, she cannot do it unless she can specify a separate variable name for the referenced key.

Reply:
I still count only the secret name and var holding the list of keys: 2

Reply:
And can we refrain from using he/she? Just "one" has always done the job

Reply:
> I still count only the secret name and var holding the list of keys: 2

I'm sorry, I don't really understand what you imply with this statement. The documentation has an example I wanted to stick to:

```yaml
env: 
      - name: SECRET_USERNAME
        valueFrom:
          secretKeyRef:
            name: mysecret
            key: username
      - name: SECRET_PASSWORD
        valueFrom:
          secretKeyRef:
            name: mysecret
            key: password
```
`mysecret` is referenced multiple times with 2 different keys. If the name and key were the same, it would look like this in this specification:

```yaml
env: 
      - name: mysecret
        valueFrom:
          secretKeyRef:
            name: mysecret
            key: username
      - name: mysecret
        valueFrom:
          secretKeyRef:
            name: mysecret
            key: password
```

I would assume this would throw a reference error or something similar.

Reply:
Ok I get it, I forgot about being able to name the var name differently than the key name. Go ahead!